### PR TITLE
Minor fixes / Make callbacks work with mode and metrics / use Conditional imports 

### PR DESCRIPTION
**Pull request by YannDubs**
Minor Fixes:
- typo in `nift_loader`, so wasn't working. 'x.endsiwth' replaced by 'x.endswith'.
- small number of changes so that `.fit_load()` works also with a `torch.utils.data.Dataset` and not only with your `TensorDataset`.
- `ModelCheckpoint`'s `max_save` was off by 1. Anoying because usually you want to keep the best model, and with the current code on master you would keep nothing! Also added an example in the docstring.

Make callbacks work with mode and metric:
- The callbacks `ModelCheckpoint`, `EarlyStopping`, `ReduceLROnPlateau` all accept a mode parameter. And it works with metrics. example : `EarlyStopping(patience=10,monitor="val_acc_metric")`

Conditional imports:
- I think there are dependencies like pandas and nibabel that are very specific that there shouldn't be required to use the package. SO i used conditional imports.

As the changes are minor I put all in the same pull request but if you want I can split.

### CODE REVIEW

**Comment by synchro--** [comment]
Is torchsample actually maintained anymore? It is a great library but maybe it'd good to expand the number of maintainers to make this project flourish.

**Comment by justiceamoh** [comment]
I agree with you @synchro-- . I'd be willing to help maintain.

**Comment by drdarina** [comment]
Same here @synchro-- @justiceamoh 
I love the functionality but I'm already running into bugs.

**Comment by YannDubs** [comment]
I also agree! And would be glad to help to maintain it: it's a shame to let this project die...

**Comment by synchro--** [comment]
Any update?
